### ISSUE TITLE
standard-minifier-js of Meteor.js remove Promise

### ISSUE THREAD

**Issue by rtalon83** [maplibre/maplibre-gl-js#4001]
A Meteor.js v2.5.8 web application using MapLibre 4.1.2 to show a simple map in a dashboard. Simple mongo links and show differents layers. In development mode all is ok.
But, When is production mode the app not run.

dist/maplibre-gl.js has this functions:

![image](https://github.com/maplibre/maplibre-gl-js/assets/12005260/5443f182-bf11-4b04-9032-8f170011be45)

Promise is changed for arbitrary variable/string, in this case 'n'

![image](https://github.com/maplibre/maplibre-gl-js/assets/12005260/888c34e3-39f3-4f90-b408-14d7c0b6c21d)

**maplibre-gl-js version**: 4.1.2

**browser**: Chrome 123.0.6312.123

### Steps to Trigger Behavior

 1. Create a new package from Meteor.js adding maplibre-gl library
 2. Set this package in packages folder.
 3. Build in production mode
 4. Map is visible but the communications are broken.

Consider using the non-minified version for this demo to create better error messages. 
E.g. `<script src='https://unpkg.com/maplibre-gl@1.15.2/dist/maplibre-gl-dev.js'></script><link href='https://unpkg.com/maplibre-gl@1.15.2/dist/maplibre-gl.css' rel='stylesheet' />`

See https://stackoverflow.com/help/mcve for guidelines on creating an effective example.
-->

**Comment by HarelM** [comment]
It is not clear what the problem is and how to reproduce it.
This looks like a bad behavior of a minifier that didn't do its job well, I'm not sure how this is related to maplibre...?
I see the word `promise` in both:
https://unpkg.com/browse/maplibre-gl@4.1.2/dist/maplibre-gl-dev.js
and 
https://unpkg.com/browse/maplibre-gl@4.1.2/dist/maplibre-gl.js

**Comment by rtalon83** [comment]
Yes, is a bad behaviour of a minifier of Meteor.js, I don't know why...
I write this issue because perhaps someone else has experienced the same thing.
Thanks.

**Comment by birkskyum** [comment]
@rtalon83 , is this still a problem with Meteor.js v3? It has e.g. replaced the fibers with native async/await support, so it could make a difference.

**Comment by rtalon83** [comment]
I have not tested my app with meteor.js v3, yet. But I resolved my problem
changing the strategy (no package, code inside the main app). My next test
will be check v3

**Comment by johncarmack1984** [comment]
A Meteor minifier bug, confirmed by the reporter. This one is resolved and can be closed.

**Comment by HarelM** [comment]
Also v6 is dramatically different, so this is not relevant anymore...
